### ISSUE TITLE
Improve spam protection in Deploy Acceptor and Block Proposer

### ISSUE THREAD

**Issue by piotr-dziubecki** [casper-network/roadmap#53]
Reject transactions that are invalid. Do not propose transactions that cannot be paid for.

**Comment by RitaMAllenCA** [comment]
@piotr-dziubecki I think this is no longer relevant AS IS, can you please either close it or rework it?  Thanks!

**Comment by RitaMAllenCA** [comment]
@marc-casperlabs is this handled in muxink? or networking 2.0? Neither?  If it is no longer relevant, please close this ticket as won't do.

**Comment by marc-casperlabs** [comment]
@RitaMAllenCA No, this is on the application level, as far as I can tell. Rejecting invalid deploys/transactions is good and makes spamming harder, regardless of the underlying networking's throttling. Also, this does apply to both networking and the RPC interface, so the impact is wider.
